backend/backend.go

```diff
@@ -150,20 +151,21 @@ func NewBackend(arguments *arguments.Arguments, environment Environment) (*Backe
 		config:      config,
 		events:      make(chan interface{}, 1000),
 
-		devices:     map[string]device.Interface{},
-		bitboxBases: map[string]bitboxbase.Interface{},
-		keystores:   keystore.NewKeystores(),
-		coins:       map[string]coin.Coin{},
-		accounts:    []accounts.Interface{},
-		log:         log,
+		devices:             map[string]device.Interface{},
+		detectedBitBoxBases: make(map[string]string),
+		bitboxBases:         map[string]bitboxbase.Interface{},
+		keystores:           keystore.NewKeystores(),
+		coins:               map[string]coin.Coin{},
+		accounts:            []accounts.Interface{},
+		log:                 log,
 	}
 	notifier, err := NewNotifier(filepath.Join(arguments.MainDirectoryPath(), "notifier.db"))
 	if err != nil {
 		return nil, err
 	}
 	backend.notifier = notifier
 
-	backend.baseManager = mdns.NewManager(backend.bitBoxBaseRegister, backend.BitBoxBaseDeregister, backend.config, backend.arguments.BitBoxBaseDirectoryPath())
+	backend.baseManager = mdns.NewManager(backend.BitBoxBaseDetect, backend.BitBoxBasesDetected, backend.bitBoxBaseRegister, backend.BitBoxBaseDeregister, backend.config, backend.arguments.BitBoxBaseDirectoryPath())
```

Comment:
feels strange to have a function callback here to give back the detected bases to the manager. The manager should already know about them, as it detected them in the first place?

Could go further and keep both the detected and registered bases only in the manager, and expose them with functions. Would mean only one variable in the backend struct `baseManager` instead of three, and probably less duplicated work (keeping track or bases).

(similar improvement maybe possible with devices as well, but that could be done later)

Reply:
Agreed, I'll change it :)

Reply:
Changed so detected bases are now kept in the manager and backend requests them with `backend.baseManager.GetDetectedBases()`

Not yet convinced about moving the registered bases to the manager. Right now the job of the manager is to run the mDNS scans and keep track of what it gets from those, not to actually initialize the Bases in the backend. 

This could definitely change, but for now I think it makes sense to register the bases with the backend.